src/ess/reduce/scicat.py

```diff
@@ -41,6 +44,7 @@ def download_scicat_file(
 ) -> FilePath:
     if target is None:
         target = Path(f'~/.cache/essreduce/{dataset_id}')
-    dset = client.get_dataset(dataset_id)
-    dset = client.download_files(dset, target=target, select=filename)
+    with _scicat_download_lock:
+        dset = client.get_dataset(dataset_id)
+        dset = client.download_files(dset, target=target, select=filename)
```

Comment:
I suppose this will break when multiple process / nodes are used that share the same filesystem?

Reply:
Yes it will not work in that situation.
If the user pre-downloads the datasets they need then `client.download_files` will not download anything (right? @jl-wynen) and then there won't be any race condition. Is that good enough?

What kind of solution would you like to see @SimonHeybrock ?

Reply:
I do not have any specific solution in mind. I just think that we should take a step back and see if we can take a different approach (such as running something *before* the pipeline) that can avoid the problem, instead of trying to fix the symptoms.

Reply:
Downloading the data before running the pipeline is already an option. The user can do that using scitacean or calling the function here directly.

I'll leave this PR as is for now until we figure out what kind of solution we want here.